src/basho_bench_worker.erl

```diff
@@ -229,14 +242,16 @@ worker_idle_loop(State) ->
             end
     end.
 
-worker_next_op2(State, OpTag) ->
-   catch (State#state.driver):run(OpTag, State#state.keygen, State#state.valgen,
-                                  State#state.driver_state).
+worker_next_op2(false, State, OpTag) ->
+    catch (State#state.driver):run(OpTag, State#state.keygen, State#state.valgen,State#state.driver_state);
+worker_next_op2(true, State, OpTag) ->
+   catch (State#state.driver):run(OpTag, State#state.driver_state, State).
```

Comment:
You've already got the `api_pass_state` value in the State record, no need to add an additional parameter here, just match directly on the State. For instance:

```
worker_next_op2(#state{api_pass_state=false}=State, OpTag) ->
    ...;
worker_next_op2(State, OpTag) ->
   ...
```